sources/engine/Xenko.Engine/Rendering/ModelRenderProcessor.cs

```diff
@@ -144,7 +144,7 @@ private void CheckMeshes(ModelComponent modelComponent, RenderModel renderModel)
                 if (model != null)
                 {
                     // Number of meshes changed in the model?
-                    if (model.Meshes.Count != renderModel.Meshes.Length)
+                    if (model.Meshes.Count != renderModel.UniqueMeshCount)
```

Comment:
From what I understood of the issue you mentioned on the post this would be resolved by comparing the two collections but you would rather not pay the perf cost of doing so ? Perhaps this could be resolved by introducing a collection with a ``version`` variable, you would test against the ``version`` variable instead of ``UniqueMeshCount`` and, if we really want to be thorough, do a collection comparison afterwards ?
If you think this would resolve the issue, @xen2 could we add a ``version`` to FastList or would you rather have another collection type for this purpose ?

Reply:
Yes, the current faulty logic is trying to detect changes in the Meshes list, I think.
If you're willing for a proper breaking change, I think changing Meshes from a `List` to a `TrackingCollection` on the `Model` and having some sort of `RequiresMeshRegen` bool will suffice.
Even if they end up adding the exact same elements, I don't think it matters too much if it regenerates the mesh (especially since they're probably going through the regen code currently and it shouldn't happen every update frame anyway).
However, I think this will need some sort of serialization upgrade code or something? All I know is Model is one of the most complicated asset types and I do not dare to mess with that, haha.

Reply:
Let's see what @xen2 has to say

Reply:
I think even if we were to compare collection, it wouldn't be enough as `Mesh` and `Model` is not immutable (which would help a lot for some specific classes such as `Mesh` to simplify a lot of those clunky checks, however it would require non-trivial changes to runtime/editor).

For example, `Mesh.MaterialIndex` could still change afterwise, as well as `Model.Materials`.
That's why the second step is not checking not the `Mesh` (either as a class reference or all its content), but only the important part requiring a rebuild (actual material with all the indirections resolved).

Reply:
Ok, perhaps this change is not safe enough to make at this point in time.
I am ok with this not included in the pull.
Do you want me to remove this commit from this pull, or leave every thing as is and you'll sort it out from the rest of the commits?

Reply:
Please remove it, I'll merge afterwards if xen doesn't do it before I have time to :)